pkg/detectors/npm/npm_token_generic.go

```diff
@@ -0,0 +1,78 @@
+package npm
+
+import (
+	"context"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type ScannerGeneric struct {
+	npmScanner
+}
+
+// Ensure the Scanner satisfies the interfaces at compile time.
+var _ interface {
+	detectors.Detector
+	detectors.Versioner
+} = (*ScannerGeneric)(nil)
+
+func (s ScannerGeneric) Version() int { return 0 }
```

Comment:
Version 0 might not be the most idiomatic way to approach this.

Reply:
iirc the way we implemented filtering depends on the version being non-zero, so you wouldn't be able to exclude/ignore just the `v0` detector.

Reply:
I've added a test [here](https://github.com/trufflesecurity/trufflehog/pull/2272) to codify that assumption.